planner/core/preprocess.go

```diff
@@ -1024,6 +1025,10 @@ func (p *preprocessor) checkContainDotColumn(stmt *ast.CreateTableStmt) {
 	}
 }
 
+func (p *preprocessor) stmtType() string {
+	return "SELECT" // TODO: This needs to return SELECT, SHOW, DELETE, INSERT, UPDATE etc.
+}
+
```

Comment:
I couldn't find an easy way to detect the statement type in the preprocessor. Looking for suggestions :-)

Reply:
I've added this the way that kind of made sense.